api/api/encoder.py

```diff
@@ -41,6 +43,8 @@ def default(self, o: object) -> dict:
             return result
         elif isinstance(o, AbstractWazuhResult):
             return o.render()
+        elif is_dataclass(o):
```

Comment:
We should add a new test case in `api/api/test/test_encoder.py` to make sure this is working as intended.